provider/resources.go

```diff
@@ -2469,6 +2469,36 @@ func ProviderFromMeta(metaInfo *tfbridge.MetadataInfo) *tfbridge.ProviderInfo {
 						AltTypes:  []tokens.Type{awsType(iamMod, "documents", "PolicyDocument")},
 						Transform: tfbridge.TransformJSONDocument,
 					},
+					"inline_policy": {
+						// inline_policy is an array of policy objects. It is allowed to provided an empty list
+						//   inlinePolicies: []
+						// or a list with empty objects
+						//   inlinePolicies: [{}]
+						// In both cases the role will be created _without_ any inline policies attached.
+						// If a policy is provided, then both the `policy` and the `name` fields are required.
+						// If one is provided and the other is not, then no error will be thrown and no inline policy
+						// will be created.
+						Transform: func(pv resource.PropertyValue) (resource.PropertyValue, error) {
+							if pv.IsArray() {
```

Comment:
This filtering here is interesting so it looks like it will drop `{}` and `{"policy": ""}`  and proceed as if they were not there. To clarify, we certainly want this behavior?

Reply:
Yeah I think this is the behavior that we want. If you provide `{"policy": ""}` it won't create the policy in AWS, but with the auto name you will end up with `{"policy": "", "name": "example-abcd"}`. This is also not valid and will end up with no policy being created in AWS (but it will show in the diff which is confusing to users).

The only valid policy is with _both_ `name` and `policy` and for backwards compatibility reasons we have to allow for users to specify `[{}]`